qcodes/instrument_drivers/Keysight/keysight_34980a_submodules.py

```diff
@@ -0,0 +1,384 @@
+                                                 'ISO'),
+                           docstring='get and set relay protection mode.')
+        self.slot = slot
+        configuration = self.ask(f'SYSTEM:MODule:TERMinal:TYPE? {self.slot}')
+        self._is_locked = (configuration == 'NONE')
+        if self._is_locked:
+            logging.warning(f'For slot {slot}, no configuration module'
+                            f'connected, or safety interlock jumper removed.')
+        else:
+            self.row, self.column = [
+                int(num) for num in re.findall(r'\d+', configuration)
+            ]
+
+    def write(self, cmd: str):
+        if self._is_locked:
+            logging.warning("Warning: no configuration module connected, "
+                            "or safety interlock enabled")
+            return None
+
+        return super().write(cmd)
+
+    def validate_value(self, row: int, column: int) -> None:
+        """
+        to check if the row and column number is within the range of the
+        module layout.
+
+        Args:
+            row (int): row value
+            column (int): column value
+        """
+        if (row > self.row) or (column > self.column):
+            raise ValueError('row/column value out of range')
+
+    @post_execution_status_poll
+    def _get_relay_protection_mode(self):
+        return self.ask(f'SYSTem:MODule:ROW:PROTection? {self.slot}')
+
+    @post_execution_status_poll
+    def _set_relay_protection_mode(self, mode: str = 'AUTO100'):
+        """
+        set the relay protection mode. The fastest switching speeds for relays
+        in a given signal path are achieved using the FIXed or ISOlated modes,
+        followed by the AUTO100 and AUTO0 modes. There may be a maximum of 200
+        Ohm of resistance, which can only be bypassed by "AUTO0" mode.
+        See manual and programmer's reference for detailed explanation.
+
+        Args:
+            mode: names for protections modes
+        """
+        self.write(f'SYSTem:MODule:ROW:PROTection {self.slot}, {mode}')
+
+    def to_channel_list(
+            self,
+            paths: List[Tuple[int, int]],
+            wiring_config: Optional[str] = None
+    ) -> str:
+        """
+        convert the (row, column) pair to a 4-digit channel number 'sxxx', where
+        s is the slot number, xxx is generated from the numbering function.
+
+        Args:
+            paths: list of channels to connect [(r1, c1), (r2, c2), (r3, c3)]
+            wiring_config (str): for 1-wire matrices, values are 'MH', 'ML';
+                                 for 2-wire matrices, values are 'M1H', 'M2H',
+                                 'M1L', 'M2L'
+
+        Returns:
+            in the format of '(@sxxx, sxxx, sxxx, sxxx)', where sxxx is a
+            4-digit channel number
+        """
+        layout = f'{self.row}x{self.column}'
+        numbering_function = self.get_numbering_function(layout, wiring_config)
```

Comment:
if using tuples of ints as keys for `available_layouts`, there will be a mypy error  `Unsupported right operand type for in ("object")` for the line:
`if wiring_config not in available_layouts[layout]:` 
(where layout is a tuple of ints)

Looks like it's a known bug:
https://github.com/python/mypy/issues/5835

However, even if I use a string for the variable `layout`, mypy still gave me the same error message. I'm trying to figure out why...

Reply:
I had row and column as input arguments, but kept layout as str for the dictionary keys. Another thing I changed is the default wiring_config: it was None, but I changed to ''. Because the following line
`if wiring_config not in available_layouts[layout]:`
kept giving me the `Unsupported right operand type for in ("object")` error, if I have [None] or [] as values of the `available_layouts`.